scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -307,45 +298,46 @@ def _get_mean_bins(
     return pd.cut(means, bins=bins)
 
 
-def _stats_seurat(mean_bins: pd.Series, disp_grouped: SeriesGroupBy) -> pd.DataFrame:
-    """Compute mean and std dev per bin."""
-    disp_bin_stats = disp_grouped.agg(avg="mean", dev=partial(np.std, ddof=1))
+def _get_disp_stats(
+    df: pd.DataFrame, flavor: Literal["seurat", "cell_ranger"]
+) -> pd.DataFrame:
+    disp_grouped = df.groupby("mean_bin", observed=True)["dispersions"]
+    if flavor == "seurat":
+        disp_bin_stats = disp_grouped.agg(avg="mean", dev="std")
+        _postprocess_seurat(disp_bin_stats, df["mean_bin"])
+    elif flavor == "cell_ranger":
+        disp_bin_stats = disp_grouped.agg(avg="median", dev=_mad)
+    else:
+        raise ValueError('`flavor` needs to be "seurat" or "cell_ranger"')
+    return disp_bin_stats.loc[df["mean_bin"]].set_index(df.index)
+
+
+def _postprocess_seurat(disp_bin_stats: pd.DataFrame, mean_bin: pd.Series) -> None:
```

Comment:
```suggestion
        _postprocess_dispersions_seurat(disp_bin_stats, df["mean_bin"])
    elif flavor == "cell_ranger":
        disp_bin_stats = disp_grouped.agg(avg="median", dev=_mad)
    else:
        raise ValueError('`flavor` needs to be "seurat" or "cell_ranger"')
    return disp_bin_stats.loc[df["mean_bin"]].set_index(df.index)


def _postprocess_dispersions_seurat(disp_bin_stats: pd.DataFrame, mean_bin: pd.Series) -> None:
```

Suggestion for a more descriptive name